### PR TITLE
build: split raidboss/oopsy data out

### PR DESCRIPTION
**Pull request by MaikoTan**

### CODE REVIEW

**Comment by quisquous** [comment]
Does this work for you locally? This has the same issue that #2293 had for me, in that if I use this branch, restart webpack dev server via `npm start`, then raidboss and oopsy don't work at all.  There's no errors, it's more that the files are not loaded at all.  If I look in the inspector, there are no files loaded from the dev server.

main:
image

patch:
image

**Comment by MaikoTan** [comment]
Looks like that we should have a explicit `import()` or `require()` so this approach would work.
Another way is to add a `script` label in html file to load these js?

**Comment by quisquous** [comment]
> Looks like that we should have a explicit `import()` or `require()` so this approach would work.
> Another way is to add a `script` label in html file to load these js?

At the moment, I don't think it's possible to add an `import` or `require` from the main html page.  This is so that cactbot can continue being loaded via `file://` urls, which doesn't support ESM.

Is there another place that the `import` can be placed?

**Comment by MaikoTan** [comment]
Sorry I found it did even not loaded the `raidboss.js`, besides, webpack complains that `raidboss.js` didn't export anything, so cause the problem.

File: ui/config/config.html [hunk 1]
```diff
@@ -4,6 +4,7 @@
   <meta charset="utf-8" />
   <link rel="stylesheet" href="../../resources/defaults.css">
   <link rel="stylesheet" type="text/css" href="config.css">
+  <script src="../../dist/raidboss_data.bundle.js"></script>
```
Comment (quisquous) [inline ui/config/config.html:7]:
```suggestion
  <script src="../../dist/raidboss_data.bundle.js"></script>
  <script src="../../dist/oopsyraidsy_data.bundle.js"></script>
```

**Review by quisquous — COMMENTED** [review]
Re: HtmlWebpackPlugin.  I think that seems reasonable, if you want to send a PR for it.  I think the only thing I'd want to make sure is that we keep all the `raidboss_alerts_only.html` redirect htmls working.

I think having something that generates the includes automatically will be much better than manual, as if you forget any shared script then nothing loads at all, and we have no testing for that.

File: ui/config/config.html (hunk 1 above)
Comment (quisquous) [reply ui/config/config.html:7]: I think with this change, this is ready to merge, in my opinion.

**Comment by MaikoTan** [comment]
> Re: HtmlWebpackPlugin.  I think that seems reasonable, if you want to send a PR for it.  I think the only thing I'd want to make sure is that we keep all the `raidboss_alerts_only.html` redirect htmls working.

Then we need an extra source directory to put the "template html", if we want to make sure all the html url would not change.
i.e: `src/pages/raidboss.html -> ui/raidboss/raidboss.html`

**Pull request merged**